### ISSUE TITLE
Types functions internal

### ISSUE THREAD

**Issue by alicevhenshaw** [airswap/airswap-protocols#226]
Currently the functions in Types.sol (our types library) are marked as external. This means we have to deploy and link the Types library for any contract that uses it. Due to the fact that the Types library is used rarely, and the number of uses of the 2 functions are once and twice respectively, it may make sense to make the functions as internal. This would mean we don't have to deploy Types.sol at all, and the contracts that utilise it can pull in what they need for deployment.

**Comment by alicevhenshaw** [comment]
Update to this:
Structs in libraries are classed as internal. This means that any contract that uses the structs merely has the structs copied within the contract before deployment, and no link to the library is required.

**Comment by poiresel** [comment]
Implementation may be less intuitive simple change to Types resulted in 
Error: CompileError: InternalCompilerError: Assembly exception for bytecode when using memory

Invalid implicit conversion from bytes memory to bytes storage pointer requested. when using storage

**Comment by poiresel** [comment]
There are two uses of the Types library, one within Swap and the other IDelegate. Each uses two separate portions of the contract.

**Comment by alicevhenshaw** [comment]
The change works if `hashDomain` is copied directly into Swap.sol, and the constant is also copied across. We've discussed this in person and concluded the complexities of the errors we got mean this change isn't worth making.

However, Types is a library that services the swap contract. The only part of Types that does not service Swap is `struct Rule`, which is used in the Delegate contract. It therefore makes sense to copy Rule across into Delegate, and keep Types as a library only for Swap.
